### PR TITLE
fix(vue3): migrate sessionIssueHandler mixin to composable

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Migrate mixin `sessionIssueHandler` to composable

### 🖼️ Screenshots / Screencasts

No visual changes, redirect to another tabs still working

>[!TIP]
>To test:
> - join conversation, then duplicate a tab in the same window -> duplicated session
> - join conversation, then remove yourself from conversation as another moderator -> deleted session 

### 🚧 Tasks

- [ ] Check sidebars

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -81,15 +81,15 @@ export default {
...
 	],
 
 	setup() {
-		const isInCall = useIsInCall()
-		const supportSessionState = useActiveSession()
-
-		return { isInCall, supportSessionState }
+		return {
+			isInCall: useIsInCall(),
+			isLeavingAfterSessionIssue: useSessionIssueHandler(),
```
Comment (ShGKme) [inline src/App.vue:90]: `isLeavingAfterSessionIssue` is not used here

File: src/composables/useSessionIssueHandler.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+	onBeforeUnmount(() => {
+		EventBus.$off('duplicate-session-detected', duplicateSessionTriggered)
+		EventBus.$off('deleted-session-detected', deletedSessionTriggered)
+	})
+
+	onBeforeMount(() => {
+		EventBus.$on('duplicate-session-detected', duplicateSessionTriggered)
+		EventBus.$on('deleted-session-detected', deletedSessionTriggered)
+	})
```
Comment (ShGKme) [inline src/composables/useSessionIssueHandler.js:47]: To follow Vue.js options recommended and logical order

```suggestion
	onBeforeMount(() => {
		EventBus.$on('duplicate-session-detected', duplicateSessionTriggered)
		EventBus.$on('deleted-session-detected', deletedSessionTriggered)
	})

	onBeforeUnmount(() => {
		EventBus.$off('duplicate-session-detected', duplicateSessionTriggered)
		EventBus.$off('deleted-session-detected', deletedSessionTriggered)
	})
```

**Review by ShGKme — APPROVED** [review]

File: src/App.vue (hunk 1 above)
Comment (Antreesy) [reply src/App.vue:90]: I'd disagree here, i see two occurences of this ref:

https://github.com/nextcloud/spreed/blob/c67f8ad543a64d8ffb37c7b53bfe5b417c46b51c/src/App.vue#L122

https://github.com/nextcloud/spreed/blob/c67f8ad543a64d8ffb37c7b53bfe5b417c46b51c/src/App.vue#L296

**Pull request merged**